pkg/epp/handlers/request.go

```diff
@@ -32,6 +32,37 @@ import (
 func (s *StreamingServer) HandleRequestHeaders(reqCtx *RequestContext, req *extProcPb.ProcessingRequest_RequestHeaders) error {
 	reqCtx.RequestReceivedTimestamp = time.Now()
 
+	// Headers must be processed first to populate the request context as subsequent logic (like body processing) may
+	// depend upon it.
+	for _, header := range req.RequestHeaders.Headers.Headers {
+		key := header.Key
+		// Per the Envoy API, a header's value can be in either the `RawValue` (bytes) or `Value` (string) field.
```

Comment:
should we fix ExtractHeaderValue then? see https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/main/pkg/epp/util/request/headers.go#L35;

Probably best if we have a function for the if/else logic that we use across the code for reading the header value and name it `ExtractHeaderValue` and rename the current `ExtractHeaderValue` to `ExtractCaseInsensitiveHeaderValue`

Reply:
Yes, this is a much cleaner solution, thanks! I missed this helper as I was tracing the request path. Most of these were discovered through hermetic test failures post-refactoring, and I am not deeply familiar with this part of the codebase. I will consolidate this into the helper.